scripts/performance/perf_traffic.py

```diff
@@ -0,0 +1,288 @@
+
+        self.args = parser.parse_args()
+
+
+def catch_number_of_request_samples():
+    """
+    Parse number of sample of "GET" requests will be created.
+    If the number is less than of equal with zero, default value (100) will be
+    returned.
+
+    :return: number of sample of "GET" requests.
+    """
+    import sys
+    result = 100
+
+    if "--init" in sys.argv:
+        index = sys.argv.index("--init")
+        if index < len(sys.argv) - 1:
+            temp = -1
+            try:
+                temp = int(sys.argv[index + 1])
+            except ValueError:
+                pass
+            if temp > 0:
+                result = temp
+
+    return result
+
+
+class TesterSimulateTraffic(Tester):
+    __sample_req_info = {}
+    __kinds_of_request = ["nym", "attribute", "schema", "claim",
+                          "get_nym", "get_attribute", "get_schema",
+                          "get_claim"]
+    __number_of_request_samples = catch_number_of_request_samples()
+
+    def __init__(self, number_of_clients: int = 2,
+                 transactions_delay: int = 100,
+                 time_out: int = 300, log=False,
+                 seed="000000000000000000000000Trustee1"):
+        super().__init__(log=log, seed=seed)
+        utils.run_async_method(
+            None, TesterSimulateTraffic._prepare_samples_for_get_req,
+            TesterSimulateTraffic.__number_of_request_samples)
+
+        if time_out <= 0 or transactions_delay <= 0 or number_of_clients <= 0:
+            return
+
+        self.transactions_delay = transactions_delay
+        self.time_out = time_out
+        self.number_of_clients = number_of_clients
+        self.current_total_txn = 0
+        self.__current_time = time.time()
+        self.__lock = threading.Lock()
+        self.__sender = requests_sender.RequestsSender()
+
+    async def _test(self):
+        """
+        Override from "Tester" class to implement testing steps.
+        """
+        lst_threads = list()
+        self.__current_time = time.time()
+        for _ in range(self.number_of_clients):
+            thread = threading.Thread(target=self.__simulate_client)
+            thread.setDaemon(True)
+            thread.start()
+            lst_threads.append(thread)
+
+        for thread in lst_threads:
+            thread.join(self.time_out * 1.1)
+
+        self.passed_req = self.__sender.passed_req
+        self.failed_req = self.__sender.failed_req
+        self.fastest_txn = self.__sender.fastest_txn
+        self.lowest_txn = self.__sender.lowest_txn
+
+    def __update(self):
+        """
+        Synchronize within threads to update some necessary information.
+        """
+        self.__lock.acquire()
+
+        if self.start_time == 0 and self.finish_time != 0:
+            self.start_time = self.finish_time
+
+        if self.current_total_txn != 0 and \
+                self.current_total_txn % self.transactions_delay == 0:
+            time.sleep(random.randint(1, 10))
```

Comment:
1) We need to be able to specify delay (random delay is just one of the possible options that also makes sense)
2) We need to be able to specify the type of request (nym, attr, claim_def, etc.). Random choosing of req type is just one of the options (which makes total sense).